### PR TITLE
Added the add repository command

### PR DESCRIPTION
**Pull request by RedbackThomson**
Description of changes:
Adds a new set of commands under the `add` verb. The first of which is `add repository` (or `add repo`) command. This will ensure the fork and clone, and then add the repository to the configuration file.

This should greatly reduce the amount of time to set up new repositories locally.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by a-hilaly** [comment]
Thanks @RedbackThomson! I'll rebase my dev branch on your PR and see if everything works as expected. Your changes might have fixed a very sneaky bug in `repository.Manager` :)

**Review by jaypipes — APPROVED** [review]
Nice :)

File: pkg/repository/manager.go [hunk 1]
```diff
@@ -276,37 +265,20 @@ func (m *Manager) EnsureFork(ctx context.Context, repo *Repository) error {
...
-
-		newPath := filepath.Join(
-			filepath.Dir(repo.FullPath),
-			repo.Name,
-		)
-		err := renameDirectory(repo.FullPath, newPath)
-		if err != nil {
-			return err
-		}
-	}
```
Comment (a-hilaly) [inline pkg/repository/manager.go:301]: why delete this block of code? wasn't it working as expected?

File: cmd/ackdev/cmd/add_repo.go [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+	Aliases: []string{"repo", "repos", "repository", "repositories"},
+	RunE:    addRepository,
+	Args:    cobra.MinimumNArgs(1),
+}
+
+func addRepository(cmd *cobra.Command, args []string) error {
+	for _, service := range args {
+		service = strings.ToLower(args[0])
+
+		ctx := cmd.Context()
```
Comment (a-hilaly) [inline cmd/ackdev/cmd/add_repo.go:42]: Oh i didn't know cobra commands have a context

File: pkg/repository/manager.go [hunk 3]
```diff
@@ -90,30 +89,14 @@ type Manager struct {
...
-	switch t {
-	case RepositoryTypeCore:
-		if !util.InStrings(name, m.cfg.Repositories.Core) {
-			return nil, ErrUnconfiguredRepository
-		}
-	case RepositoryTypeController:
-		if !util.InStrings(name, m.cfg.Repositories.Services) {
-			return nil, ErrUnconfiguredRepository
-		}
-	}
```
Comment (a-hilaly) [inline pkg/repository/manager.go:110]: unfortunately this have broken some other commands/tests - sending a follow-up PR

File: pkg/repository/manager.go (hunk 1 above)
Comment (RedbackThomson) [reply pkg/repository/manager.go:301]: I just cloned directly into `repo.FullPath` instead. There was no need to clone it with the fork as the name and then rename it.

**Pull request merged**